src/main/java/software/amazon/awssdk/crt/http/Http2StreamManager.java

```diff
@@ -0,0 +1,282 @@
+package software.amazon.awssdk.crt.http;
+
+import software.amazon.awssdk.crt.CrtResource;
+import software.amazon.awssdk.crt.CrtRuntimeException;
+import software.amazon.awssdk.crt.io.ClientBootstrap;
+import software.amazon.awssdk.crt.io.SocketOptions;
+import software.amazon.awssdk.crt.AsyncCallback;
+import software.amazon.awssdk.crt.io.TlsContext;
+
+import java.util.concurrent.CompletableFuture;
+import java.net.URI;
+import java.nio.charset.Charset;
+
+/**
+ * Manages a Pool of HTTP/2 Streams. Creates and manages HTTP/2 connections
+ * under the hood.
+ */
+public class Http2StreamManager extends CrtResource {
+
+    private static final String HTTP = "http";
+    private static final String HTTPS = "https";
+    private static final int DEFAULT_HTTP_PORT = 80;
+    private static final int DEFAULT_HTTPS_PORT = 443;
+    private final static Charset UTF8 = java.nio.charset.StandardCharsets.UTF_8;
+
+    private final URI uri;
+    private final int port;
+    private final int maxConnections;
+    private final int idealConcurrentStreamsPerConnection;
+    private final int maxConcurrentStreamsPerConnection;
+    private final CompletableFuture<Void> shutdownComplete = new CompletableFuture<>();
+
+    /**
+     * Factory function for Http2StreamManager instances
+     *
+     * @param options configuration options
+     * @return a new instance of an Http2StreamManager
+     */
+    public static Http2StreamManager create(HttpStreamManagerOptions options) {
```

Comment:
not a fan of the HTTP/2 manager taking generic "HTTP" options.
every time we try and make a generic all purpose thing we regret it
let's have the HTTP/2 manager take HTTP/2 options
and an HTTP/1 manager can take HTTP/1 options
or there can be some options base class for common params
or if there's no HTTP/1-specific options then HTTP/1 can use the base class and HTTP/2 can use the HTTP/2-specific class